src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier randomColor = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        Name[] names = Name.values();
+        int index = random.nextInt(names.length);
+        if (names[index].toString().equals("SQUARE")) {
+            return new Square(
+                     random.nextDouble(),
+                    randomColor.getRandomColor(),
+                    names[index].toString()
+            );
+        }
+        if (names[index].toString().equals("RECTANGLE")) {
+            return new Rectangle(
+                          random.nextDouble(),
+                          random.nextDouble(),
+                          randomColor.getRandomColor(),
+                          names[index].toString()
+                      );
+        }
+        if (names[index].toString().equals("RIGHT_TRIANGLE")) {
+            return new RightTriangle(
+                    random.nextDouble(),
+                    random.nextDouble(),
+                    randomColor.getRandomColor(),
+                    names[index].toString()
+            );
+        }
+        if (names[index].toString().equals("CIRCLE")) {
+            return new Circle(
+                    random.nextDouble(),
+                    randomColor.getRandomColor(),
+                    names[index].toString()
+            );
+        } else {
+            return new IsoscelesTrapezoid(
+                    random.nextDouble(),
+                    random.nextDouble(),
+                    random.nextDouble(),
+                    randomColor.getRandomColor(),
+                    names[index].toString()
+            );
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(10,"WHITE","CIRCLE");

```

Comment:
Don't return `null` from a method. Instead of using a separate method for getting a default figure, consider using the `getDefaultFigure()` method within the `getRandomFigure()` method in case of a specific condition, or handling the default case within the `switch` or `if-else` logic of `getRandomFigure()`.